src/components/subtitlesync/subtitlesync.js

```diff
@@ -1,125 +1,142 @@
-        }, 0);
-    }
-
-    subtitleSyncContainer.classList.add('hide');
-
-    subtitleSyncTextField.updateOffset = function (offset) {
-        this.textContent = offset + 's';
-    };
-
-    subtitleSyncTextField.addEventListener('click', function () {
-        // keep focus to prevent fade with osd
-        this.hasFocus = true;
-    });
-
-    subtitleSyncTextField.addEventListener('keydown', function (event) {
-        if (event.key === 'Enter') {
-            // if input key is enter search for float pattern
-            let inputOffset = /[-+]?\d+\.?\d*/g.exec(this.textContent);
-            if (inputOffset) {
-                inputOffset = inputOffset[0];
-                inputOffset = parseFloat(inputOffset);
-
-                subtitleSyncSlider.updateOffset(inputOffset);
-            } else {
-                this.textContent = (playbackManager.getPlayerSubtitleOffset(player) || 0) + 's';
-            }
-            this.hasFocus = false;
-            event.preventDefault();
-        } else {
-            // keep focus to prevent fade with osd
-            this.hasFocus = true;
-            if (event.key.match(/[+-\d.s]/) === null) {
-                event.preventDefault();
-            }
-        }
+// Constants
+const DEFAULT_OFFSET = 0;
+
+function createSliderBubbleHtml(_, value) {
+    return '<h1 class="sliderBubbleText">'
+        + (value > 0 ? '+' : '') + parseFloat(value) + 's'
+        + '</h1>';
+}
 
-        // FIXME: TV layout will require special handling for navigation keys. But now field is not focusable
-        event.stopPropagation();
-    });
+class OffsetController {
+    constructor(player, slider, textField) {
+        this.player = player;
+        this.slider = slider;
+        this.textField = textField;
 
-    subtitleSyncTextField.blur = function () {
-        // prevent textfield to blur while element has focus
-        if (!this.hasFocus && this.prototype) {
-            this.prototype.blur();
-        }
-    };
+        this._initSlider();
+        this._initTextField();
+
+        // Set initial offset
+        this.reset();
+    }
+
+    get currentOffset() {
+        return parseFloat(this.slider.value);
+    }
+
+    set currentOffset(value) {
+        this.slider.value = value.toString();
 
-    function updateSubtitleOffset() {
-        const value = parseFloat(subtitleSyncSlider.value);
-        // set new offset
-        playbackManager.setSubtitleOffset(value, player);
-        // synchronize with textField value
-        subtitleSyncTextField.updateOffset(value);
+        // rely on slider value trimming
+        value = this.currentOffset;
+
+        playbackManager.setSubtitleOffset(value, this.player);
+        this.textField.updateOffset(value);
     }
 
-    subtitleSyncSlider.updateOffset = function (sliderValue) {
-        // default value is 0s = 0ms
-        this.value = sliderValue === undefined ? 0 : sliderValue;
+    _initSlider() {
```

Comment:
Small code style feedback: we should use actual private methods instead of the pseudo standard underscore prefix.

Also for future reference, [we don't use "conventional" commits](https://jellyfin.org/docs/general/contributing/development#pull-request-guidelines).

Reply:
Ah, not a JS native (coming from Python), didn't even know JS had special syntax for private methods, good to learn. I made the change


For the commit, I am used to a squash merge and changed the name of the PR to non conventional, hopefully that is good.